### ISSUE TITLE
Fullscreen client without resizing?

### ISSUE THREAD

**Issue by Donaim** [awesomeWM/awesome#3107]
Please add support for optional fullscreen mode that mimics default fullscreen mode of xmonad and ratpoison. In those window managers, when non-floating clients request fullscreen, they do not resize their window.

Output of `awesome --version`:
```
awesome v4.3 (Too long)
 • Compiled against Lua 5.2.4 (running with Lua 5.2)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.6
 • LGI version: 0.9.2
```

**How to reproduce the issue:**

- switch to tiling layout
- launch firefox
- press F11

**Actual result:**

- firefox window takes the whole screen

**Expected result:**

- firefox window does not resize, but client should act as if it is fullscreen (so, in case of firefox, browser tabs should hide)

**Comment by Elv13** [comment]
Hi,

Duplicate of https://github.com/awesomeWM/awesome/pull/2429

It's technically already possible in v4.3. You can disconnect `awful.ewmh.merge_maximization` and it will mostly work. https://github.com/awesomeWM/awesome/blob/v4.3/lib/awful/ewmh.lua#L434

The PR linked above would offer a `size_hints_honor` like "thing" to make it easy in the rules. Part of that system was merged earlier this year as part of the "permission framework" project. With `git-master`, it is possible to deny that request and it should do what you describe (untested). This is still not super well documented since the second half of that system is not merged yet.

Closing since this is "done enough" and has the missing bits already covered in another issue.

**Comment by Donaim** [comment]
Hi @Elv13 and thanks for help!
As you suggested, I tried to disconnect `merge_maximization` and it almost worked - firefox did not resize, BUT another window dissapeared under firefox window, as if it became floating. Same effect has disconnecting the `ewmh.geometry` signal, which I tried earlier.

Disconnecting the signal will work for me if fixed, but I would love to see your patch merged since it provides much cleaner solution
